receiver/prometheusreceiver/config.go

```diff
@@ -165,84 +134,49 @@ func validatePromConfig(promConfig *promconfig.Config) error {
 	return nil
 }
 
-func (cfg *Config) validateTargetAllocatorConfig() error {
-	// validate targetAllocator
-	targetAllocatorConfig := cfg.TargetAllocator
-	if targetAllocatorConfig == nil {
-		return nil
-	}
-	// ensure valid endpoint
-	if _, err := url.ParseRequestURI(targetAllocatorConfig.Endpoint); err != nil {
-		return fmt.Errorf("TargetAllocator endpoint is not valid: %s", targetAllocatorConfig.Endpoint)
-	}
-	// ensure valid collectorID without variables
-	if targetAllocatorConfig.CollectorID == "" || strings.Contains(targetAllocatorConfig.CollectorID, "${") {
-		return fmt.Errorf("CollectorID is not a valid ID")
-	}
+// PromHTTPSDConfig is a redeclaration of the promHTTP.SDConfig because we need custom unmarshaling
+// because prometheus "config" uses `yaml` tags.
```

Comment:
[Nit]
```suggestion
// PromHTTPSDConfig is a redeclaration of promHTTP.SDConfig because we need custom unmarshaling
// as prometheus "config" uses `yaml` tags.
```

Reply:
Done.